verifier/testreport.py

```diff
@@ -921,6 +927,28 @@ def check_simple_text_diffs(self, test_list, category):
 
         return dict(results)
 
+    def check_list_differences(self, test_list, category):
+        #TODO Check if expected and results are differences in list content
+        results = defaultdict(list)
+        all_checks = ['different lengths', 'different_content', 'other list difference']
+        for check in all_checks:
+            results[check] = set()
+
+        for fail in test_list:
+            label = fail['label']
+            actual = fail['result']
+            expected = fail.get('expected', None)
+            if type(actual) != type(expected):
+                # This is a type mismatch
+                continue
+            if type(actual) == list:
```

Comment:
If this list-specific `check_list_differences` is kept, then this condition should go at the callsite, not here.

Reply:
See above.

Reply:
restructuring this